CMSgov/mint-app
# Easi 2607/collaborator user table migration

# EASI-2607 Plan Collaborator to User Table

## Changes and Description

- Audit Change Tables
    - Modified to use the user table conditionally for the if_modified() Trigger
    - Modified for all records to be a reference to the user account table
- Plan Collaborator to user table 
    - Plan Collaborator table is now using user table for user_id, created_by, and modified_by
    - Email and full name is now coming from the user_table
- Analyzed Audit tables
    - Plan collaborator stores user id on creation
    - model_lead looks up common name when getting a collaborator by user_id
    - Email Digest
        -  now uses user id for favorites
        - uses user id for sending email.



## How to test this change
1. Launch the front end code, and add plan collaborators.
2. Ensure that the app displays collaborators correctly and functions as expected
3. Test that the digest email sends as expected with Faktory, and sends to the correct email address now referenced by User Account.



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
